### PR TITLE
fixes getting the userId with the x-mtok from Meteor.server.sessions

### PR DESCRIPTION
**Pull request by menelike**
- Changes were introduced in https://github.com/meteor/meteor/commit/79ae1849f5846e207ed4dfa49831ba409252fd03
- Without this, this project is incompatible with Meteor >= 1.8.1
- Meteor.server.sessions is a Map and not an Object anymore.

fixes https://github.com/VeliovGroup/Meteor-Files/issues/681

### CODE REVIEW

**Comment by tschneid** [comment]
Thanks for your effort! After having a short look at the changes, wouldn't this introduce the same problems for older Meteor versions (<1.8.1)?

**Comment by menelike** [comment]
@tschneid Thanks for your feedback, initially I had a major version bump in mind, but I agree, staying backwards compatible is the better way.

*Note: I left the if clauses as they were in logical terms, even when I think they're checking things too often*

**Comment by jamalcoder** [comment]
thanks @menelike i updated my app to 1.8.1 and i think we will wait a long time to see this get merged with master branch a long time i did not see @dr-dimitru

**Comment by dr-dimitru** [comment]
Hey, going to check and sort out all pending tickets this week. I'm very sorry for being away.

**Comment by tschneid** [comment]
Any updates on this, @dr-dimitru? :slightly_smiling_face:

**Comment by jchristman** [comment]
@dr-dimitru -- this is still an issue as far as I can tell. Any chance we could get this merged sooner than later? I'm having to downgrade to an older version of meteor because of it

File: server.js [hunk 1]
```diff
@@ -1022,6 +1022,29 @@ export class FilesCollection extends FilesCollectionCore {
...
+   * @locus Anywhere
+   * @memberOf FilesCollection
+   * @name _getUserId
+   * @summary Returns `userId` matching the xmtok token derived from Meteor.server.sessions
+   * @returns {String}
+   */
+  _getUserId(xmtok) {
+    if (!xmtok || !helpers.isObject(Meteor.server.sessions)) return null;
+
+    if (Meteor.server.sessions instanceof Map
```
Comment (jankapunkt) [inline server.js]: Could you please add semicolons in the statements below, so we keep the formatting consistent.

File: server.js [hunk 2]
```diff
@@ -1022,6 +1022,29 @@ export class FilesCollection extends FilesCollectionCore {
...
+      && Meteor.server.sessions.has(xmtok)
+      && helpers.isObject(Meteor.server.sessions.get(xmtok))
+    ) {
+      return Meteor.server.sessions.get(xmtok).userId
+    } else if (helpers.isObject(Meteor.server.sessions)
```
Comment (jankapunkt) [inline server.js]: Please also add a one-liner that describes which of the both branches in the if/else handles which of the cases, for instance:

```javascript

if (Meteor.server.sessions instanceof Map ...) {
  // use with >= Meteor 1.8.1
} else if (helpers.isObject(Meteor.server.sessions) {
  // use with < Meteor 1.8.1
}

File: server.js [hunk 3]
```diff
@@ -1022,6 +1022,29 @@ export class FilesCollection extends FilesCollectionCore {
...
+      && xmtok in Meteor.server.sessions
+      && helpers.isObject(Meteor.server.sessions[xmtok])
+    ) {
+      return Meteor.server.sessions[xmtok].userId
+    }
```
Comment (jankapunkt) [inline server.js]: Could you please add an `} else {` and throw an Error there to cover this undefined state, too. Even this is highly unlikely it allows us to detect breaking changes on this in future Meteor versions that are related to this issue fast and precise.

**Comment by jankapunkt** [comment]
@jchristman this will be merged ASAP in the meantime you can fork the PR's branch from https://github.com/risetechnologies/Meteor-Files/tree/fixMeteor1.8.1 into your `packages` folder and remove it, once it is merged.

**Comment by jamalcoder** [comment]
@jchristman do as @jankapunkt says.

but let me add some notes, for users who use heroku for example, this will not work until you add your package as submodule.
inside your packages folder run:
`git submodule add -b fixMeteor1.8.1 https://github.com/risetechnologies/Meteor-Files.git`
than remove the package and install it again(in local dev)

**Comment by jankapunkt** [comment]
@jamalcoder  is this workaround required on any setup using heroku? If so it would be great if you add this as an own issue and we can discuss further actions, for example updating the docs.

**Comment by jamalcoder** [comment]
@jankapunkt yes i think it's required for heroku. any meteor package added in packages folder must be a sub module

**Pull request merged**